events/processor/index.js

```diff
@@ -0,0 +1,77 @@
+import AWS from 'aws-sdk';
+import uuid from 'uuid';
+
+import Echo from './types/echo';
+
+export default class Processor {
+  constructor(event) {
+    this.kinesis = new AWS.Kinesis();
+    this.event = event;
+    this.types = {
+      'echo': Echo
+    };
+  }
+
+  process(callback) {
+    if (!this.event.hasOwnProperty('Records') ||
+        !Array.isArray(this.event.Records)) {
+      throw "event.Record is missing; this is bad!";
+    }
+    
+    this.event.Records.forEach((record) => {
+      try {
+        const data = this.getData(record);
+        const processor = new this.types[data.type](data);
+        processor.process();
+      } catch (e) {
+        this.badEvent(e, record);
+      }
+    });
+
+    if (typeof callback === 'function') {
+      callback();
```

Comment:
I've been told by folks smarter than me on NodeJS not to mix callbacks and Promises. Lambda, for better or worse, uses callbacks in the native API. I'm indifferent for the most part; though the Promise API is a lot nicer.